pyproject.toml

```diff
@@ -5,3 +5,12 @@ skip = ["tests/imports.pyi"]
 
 [tool.black]
 force-exclude = ".*\\.pyi"
+
+[tool.mypy]
+show_error_codes = true
+check_untyped_defs = true
+disallow_any_generics = true
+warn_redundant_casts = true
+warn_return_any = true
+warn_unreachable = true
+warn_unused_ignores = true
```

Comment:
Just noticed that this file doesn't have a trailing newline. It's not a big deal, but we could also add a `.editorconfig` file in a separate PR.